ideascube/serveradmin/catalog.py

```diff
@@ -559,14 +559,21 @@ def upgrade_packages(self, ids):
 
     # -- Manage local cache ---------------------------------------------------
     def _load_catalog(self):
-        if os.path.exists(self._catalog_cache):
+        self._catalog = {'available': {}, 'installed': {}}
+
+        try:
             with open(self._catalog_cache, 'r') as f:
-                self._catalog = yaml.safe_load(f.read())
+                catalog = yaml.safe_load(f.read())
 
-        # yaml.safe_load returns None for an empty file.
-        if not getattr(self, '_catalog', None):
-            self._catalog = {'installed': {}, 'available': {}}
-            self._persist_catalog()
+            # yaml.safe_load returns None for an empty file
```

Comment:
Ooooooooh! **That** `else` block!

I thought you meant a new one for the `if`, not one for the `try`!

I see what you mean, yes. And indeed, that's probably better.